guvnor-asset-mgmt/guvnor-asset-mgmt-client/src/main/java/org/guvnor/asset/management/client/editors/repository/wizard/CreateRepositoryWizard.java

```diff
@@ -144,19 +157,26 @@ public int getPreferredWidth() {
     @Override
     public void isComplete( final Callback<Boolean> callback ) {
 
-        callback.callback( true );
+        final int[] unCompletedPages = {this.pages.size()};
+        final boolean[] completed = {false};
 
         //only when all pages are complete we can say the wizard is complete.
         for ( WizardPage page : this.pages ) {
             page.isComplete( new Callback<Boolean>() {
                 @Override
                 public void callback( final Boolean result ) {
-                    if ( Boolean.FALSE.equals( result ) ) {
-                        callback.callback( false );
+
+                    if ( Boolean.TRUE.equals( result ) ) {
+                        unCompletedPages[0]--;
+                        if ( unCompletedPages[0] == 0 ) {
+                            completed[0] = true;
+                        }
                     }
                 }
             } );
         }
+
+        callback.callback( completed[0] );
```

Comment:
Unfortunately this is likely to be called before all pages have responded to the callback. The ```isPageComplete``` method uses a callback mechanism in case the page needs to make an asynchronous call to the server (e.g. to validate a GAV component etc). Therefore ```completed[0]``` may not have been set correctly in the callbacks and may be ```false``` whereas it should be ```true```. You're wanting to use ```unCompletedPages``` as a semaphore to signal when all pages' callbacks have ran. You might want to "wait" to invoke the callback until ```unCompletedPages[0] == 0``` (and possibly time-out - with a ```Timer``` - to prevent infinite looping).